website/docs/language/import/generating-configuration.mdx

```diff
@@ -0,0 +1,123 @@
+---
+page_title: Import - Generating configuration
+description: >-
+  Import resources into Terraform to bring them under Terraform's management by writing `import` blocks in code.
+---
+
+# Import: Generating configuration
+
+-> **Note:** Config generation is available in Terraform v1.5 as an experimental feature. Later minor versions may contain changes to the format of generated configuration and behaviour of the `terraform plan` command when the `-generate-config-out` flag is supplied.
+
+Terraform can generate HCL for resources which do not already exist in configuration. The HCL produced by Terraform is intended to act as a template, and contains Terraform's best guess at the appropriate value for each resource argument. You may find that you arrive at an ideal configuration by starting from the HCL generated by Terraform, removing some attributes, adjusting the value of others, and rearranging `resource` blocks into files and modules as appropriate.
+
+To instruct Terraform to generate configuration during the plan, supply a file path with `-generate-config-out`. The file should not already exist. For example:
+
+```
+terraform plan -generate-config-out=generated_resources.tf
+```
+
+Terraform will generate configuration for any resources targeted by an `import` block that do not already exist in configuration, and write the HCL to `generated_resources.tf`.
+
+## Workflow
+
+Terraform generates configuration during the plan and writes the new HCL file to disk after the plan has completed.
+
+### 1. Add the `import` block to configuration
+
+```hcl
+provider "aws" {
+  region = "eu-west-1"
+}
+
+import {
+  to = aws_iot_thing.bar
+  id = "foo"
+}
+```
+
+### 2. Plan the import
+Run `terraform plan -generate-config-out=generated.tf`. You should see output like the following:
+
+
+```
+$ terraform plan
+
+aws_iot_thing.bar: Preparing import... [id=foo]
+aws_iot_thing.bar: Refreshing state... [id=foo]
+
+Terraform will perform the following actions:
+
+  # aws_iot_thing.bar will be imported
+    resource "aws_iot_thing" "bar" {
+        arn               = "arn:aws:iot:eu-west-1:1234567890:thing/foo"
+        attributes        = {}
+        default_client_id = "foo"
+        id                = "foo"
+        name              = "foo"
+        version           = 1
+    }
+
+Plan: 1 to import, 0 to add, 0 to change, 0 to destroy.
+```
+
+### 3. Review the generated code
+
+Open `generated.tf` in your editor and review the generated code. It should look like this:
+
+```hcl
+resource aws_iot_thing "bar" {
+  name = "foo"
+}
+```
+
+Add or remove resource arguments as desired. Now is an appropriate time to add variables or references to other resources.
+
+### 4. Apply the import
+Run `terraform apply`. You should see output like the following:
```

Comment:
I don't feel like we need the below code block, it doesn't show anything new so I think we can remove it and shorten this section to just:
```suggestion
Running `terraform apply` create and provisions your infrastructure. Ensure you add your generated code alongside your Terraform code to your version control system.
```